File: 1302_deepestLeavesSum.ts

function deepestLeavesSum(root: TreeNode | null): number {
    var maxlevel = -1;
    var sum = 0;
    function dfs(root: TreeNode | null, level: number )
    {
        if(!root) return;
        if(level > maxlevel)
        {
            maxlevel = level;
            sum = root.val;
        }else if(level == maxlevel)
        {
            sum += root.val;
        }
        dfs(root.left, level + 1);
        dfs(root.right, level + 1 );
    }
    dfs(root, 0);
    return sum;
};

File: 1302_deepestLeavesSum.cpp
#include ".\self_definition\TreeNode.h"
#include <iostream>
#include <queue>
#include <vector>
#include <numeric>
using namespace std;

class Solution
{
public:
    int deepestLeavesSum(TreeNode *root)
    {
        queue<TreeNode *> que;
        if (!root)
            return root->val;
        que.push(root);
        vector<vector<int>> res;
        while (!que.empty())
        {
            int n = que.size();
            vector<int> temp;
            for (int i = 0; i < n; i++)
            {
                TreeNode *node = que.front();
                temp.emplace_back(node->val);
                que.pop();
                if (node->left)
                    que.emplace(node->left);
                if (node->right)
                    que.emplace(node->right);
            }
            res.emplace_back(temp);
        }
        int m = res.size() - 1;
        return accumulate(res[m].begin(), res[m].end(), 0);
    }
};